### DISCUSSION TITLE
Is it possible to performance test /Stress test from Thunder client.

### DISCUSSION THREAD

**Discussion by RashmiBK5** [thunderclient/thunder-client-support#1222] [General]
I want to send 10 request at a time and see if the API is able to handle request or not.

```js
function makeRequest() {
    return new Promise((resolve, reject) => {
        fetch("{API-URL}", { method: "GET", })
            .then((respose) => respose.json())
            .then((data) => resolve(data))
            .catch((error) => reject(error));
    });
}

for (let i = 0; i < numberOfUsers; i++) {
    requests.push(makeRequest());
    console.log(i);
}

Promise.all(requests)
    .then((respose) => {
        respose.forEach((reposnse, index) => {
            console.log(`user${index + 1}:`, reposnse)
        })
    })
    .catch((error) => {
        console.error('Error:', error);
    })
```

**Comment by rangav** [comment]
Hi @RashmiBK5 currently not possible.

Is it similar to this feature request? - #1088

**Reply by RashmiBK5** [reply]
@rangav 
yes, it looks like same issue. 
But I am trying to achieve by adding the script in pre-script or post-request script.

I am trying to follow attached [document](https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#send-requests-from-script).

1. **_Execute existing requests using the API_** : I am not completely sure how to use it, what is reqId ?
2. **_Execute custom requests using the [Axios library](https://axios-http.com/docs/api_intro):_** When I trying to run Api using axio, I am getting error "unable to get local issuer certificate ".
     - I am trying to access local hosted api, I don't have any certificate, is it possible to disable certificate. 


![image](https://github.com/rangav/thunder-client-support/assets/96167898/585c1a94-a528-47b8-873c-918e8bc163fd)

**Reply by rangav** [reply]
I did not test with the scripts. but you can try it.

The request id is available in the collection json file.

![image](https://github.com/rangav/thunder-client-support/assets/8637550/afe6ed1c-5842-4fc4-b4e3-0a15d2c6026d)

**Comment by rangav** [comment]
Hi @RashmiBK5 We launched Run Collection Requests in Parallel
https://github.com/rangav/thunder-client-support/releases/tag/v2.15.0

Can you please test and let me know feedback
